src/storage-controller/src/lib.rs

```diff
@@ -2558,7 +2570,11 @@ where
 
     /// Effectively truncates the source status history shard except for the most recent updates
     /// from each ID.
-    async fn partially_truncate_status_history(&mut self, collection: IntrospectionType) {
+    /// Returns a map with the timestamp of the latest row per id
+    async fn partially_truncate_status_history(
+        &mut self,
+        collection: IntrospectionType,
+    ) -> Option<BTreeMap<GlobalId, DateTime<Utc>>> {
```

Comment:
I need to convince myself this is safe, but I also think it will be helpful for me in the future, for source/sink statuses

Reply:
however, I think you should just return the `Row` as the value, and let the callee interpret it

Reply:
> however, I think you should just return the Row as the value, and let the callee interpret it

I'll refactor to do this though it adds a slight performance penalty of an additional clone and re-packing of the row vector. I'm not sure if that's a big deal or not, since these status tables should be quite small.

Reply:
will probably be fine!